acra/security-controls/zones.md

```diff
@@ -0,0 +1,146 @@
+```
+AcraServer will process row by row, column by column. At first, it will try to match ZoneID until success.
+1. Try match `1`. It will fail because of invalid ZoneID begin tag `DDDDDDDD`.
+2. Try match `DDDDDDDDjKjECtcRBDkmHVBh` and if finds keys related with this ZoneID then it will remember until finish process row data. After that starts to match AcraStruct or AcraBlock signatures.
+3. Try match `\x222222222222222255454332....` and recognize AcraStruct signature. After that loads Zone's private key and tries to decrypt AcraStructs. On success AcraServer will replace AcraStruct with decrypted data.
+
+#### Method 2
+
+At this case application will not store ZoneID in database and store it locally or get it from user's input. Our table in database will be changed.
+For PostgreSQL:
+```
+CREATE TABLE application_data (
+ id SERIAL PRIMARY KEY,
+ data BYTEA
+);
+INSERT INTO application_data (data) VALUES ('\x2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103'::BYTEA);
+```
+
+For MySQL:
+```
+CREATE TABLE application_data (
+ id INTEGER PRIMARY KEY AUTO_INCREMENT,
+ data BLOB
+);
+INSERT INTO application_data (data) VALUES (X'2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103');
+```
+
+Now application should explicitly specify ZoneID in SQl query (works for MySQL and PostgreSQL): `SELECT "application_data"."id", 'DDDDDDDDjKjECtcRBDkmHVBh', "application_data"."data" FROM "application_data";`
+Here we placed ZoneID as string literal before encrypted column `data` and ZoneID value will be returned by database before encrypted data. So result rows will look similar as in example of [method 1](#method-1):
+```
+1 DDDDDDDDjKjECtcRBDkmHVBh \x2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103 
+```
```

Comment:
After this will be nice to put the same explanation as for the method 1. Or rather, just refer to it.

```markdown
Just like with the previous method, AcraServer will match the ZoneID in response and use it to decrypt AcraStruct data that follows in the next column.
```

Reply:
added